File: ia_helper.py
#!/bin/python3

import sys
import os
import re

if len(sys.argv) != 3:
  print ('usage: ia_helper.py archive input_file.txt')
  sys.exit()

r = re.compile(r"^\s+|\n+|\r+")

with open(sys.argv[2]) as file:
    for line in file:
        line = r.sub("", line)

        if line == "":
            continue

        print("GET: " + line)
        cmd = "ia download " + sys.argv[1] + " \"" + line + "\""
        if os.system(cmd) != 0:
            sys.exit()

File: ia_helper.pl
#!/bin/perl
use strict;
use warnings;

if(scalar @ARGV != 2)
{
  die "usage: ia_helper.pl archive inputfile.txt";
}

my ($archive, $file) = @ARGV;

open(FH, '<', $file) or die "$!";
my @rows = <FH>;
chomp @rows;
close FH;

foreach my $row (@rows)
{
  $row =~ s/^\s+|\s+$//g;
  next if $row eq "";

  print "GET: $row\n";
  system("ia", "download", $archive, $row) == 0 or die "$!";
}
